xla/service/gpu/dot_algorithm_support_test.cc

```diff
@@ -143,8 +159,23 @@ TEST_P(DotAlgorithmSupportTest, AlgorithmIsSupportedFromCudaCapability) {
       primitive_util::LowercasePrimitiveTypeName(params.output_storage_type),
       params.sizes.contracting_size, params.sizes.non_contracting_size);
 
-  if (GetCudaComputeCapability().IsAtLeast(params.min_cuda_capability.major,
-                                           params.min_cuda_capability.minor)) {
+#ifdef GOOGLE_CUDA
+  bool is_algorithm_supported =
+    GetCudaComputeCapability().IsAtLeast(params.min_cuda_capability.major,
+                                         params.min_cuda_capability.minor);
+#elif TENSORFLOW_USE_ROCM
+  bool is_algorithm_supported =
+    GetRocmComputeCapability().gfx9_mi100_or_later();
+    if (TF_ROCM_VERSION < params.min_rocm_version &&
+        (params.input_storage_type == F8E5M2 || params.input_storage_type == F8E4M3FN) &&
+        params.output_storage_type == BF16) {
+       GTEST_SKIP() << "TODO: Unsupported F8 to BF16 in ROCm version < 6.2";
+    }
+    if (params.backend_restriction == BackendRestriction::kTritonOnly) {
+        GTEST_SKIP() << "TODO: Triton unsupported in ROCm";
+    }
+#endif //GOOGLE_CUDA
```

Comment:
Is it possible to do everything without ifdefs? Just get the gpu_compute_capability() and check if it holds a cuda or rocm capability.

Reply:
This comment is marked as resolved, but I still see a lot of `#ifdef GOOGLE_CUDA` and `#elif TENSORFLOW_USE_ROCM`. These should not be needed. Please check the compute capability using run-time methods, not macros.

Reply:
@dimitar-asenov  I am not sure if I can get rid of those `#ifdefs`. In ROCm we do not store ROCM_VERSION as part of device descriptor unlike CUDA. We use TF_ROCM_VERSION which is defined in rocm_configs.h file. I plan to change this but in a different commit.

Reply:
@hsharsha Please see the discussion and solution in https://github.com/openxla/xla/pull/13479. Let's wait for that to land then you can start using the same solution.

Reply:
That PR has landed, so you should be able to use that solution here.

Reply:
@dimitar-asenov  Thanks for pointing me to that. I have updated this PR on the same lines of #13479

Reply:
NP, I left a small comment and then I will merge this. Sorry for the long back and forth.